protobuf/module_msg.proto

```diff
@@ -6,385 +6,869 @@ package bess.pb;
+ */
 message UpdateCommandClearArg {
 }
 
+/**
+ * The module WildcardMatch has a command `add(...)` which inserts a new rule into the WildcardMatch module. For an example of code using WilcardMatch see `bess/bessctl/conf/samples/wildcardmatch.bess`.
+ */
 message WildcardMatchCommandAddArg {
-  uint64 gate = 1;
-  int64 priority = 2;
-  repeated uint64 values = 3;
-  repeated uint64 masks = 4;
+  uint64 gate = 1; /// The gate to direct traffic matching this rule to.
+  int64 priority = 2; ///If a packet matches multiple rules, the rule with higher priority will be applied. If priorities are equal behavior is undefined.
+  repeated uint64 values = 3; /// The values to check for in each fieild.
+  repeated uint64 masks = 4; /// The bitmask for each field -- set 0x0 to ignore the field altogether.
 }
 
+/**
+ * The module WildcardMatch has a command `delete(...)` which removes a rule -- simply specify the values and masks from the previously inserted rule to remove them.
+ */
 message WildcardMatchCommandDeleteArg {
-  repeated uint64 values = 1;
-  repeated uint64 masks = 2;
+  repeated uint64 values = 1; /// The values being checked for in the rule
+  repeated uint64 masks = 2; /// The bitmask from the rule.
 }
 
+/**
+ * The function `clear()` for WildcardMatch takes no parameters, it clears
+ * all state in the WildcardMatch module (is equivalent to calling delete for all rules)
+ */
 message WildcardMatchCommandClearArg {
 }
 
+/**
+ * For traffic which does not match any rule in the WildcardMatch module,
+ * the `setDefaultGate(...)` function specifies which gate to send this extra traffic to.
+ */
 message WildcardMatchCommandSetDefaultGateArg {
   uint64 gate = 1;
 }
 
+/**
+ * The module ACL creates an access control module which by default blocks all traffic, unless it contains a rule which specifies otherwise.
+ * Examples of ACL can be found in [acl.bess](https://github.com/NetSys/bess/blob/master/bessctl/conf/samples/acl.bess)
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: 1
+ */
 message ACLArg {
+  /**
+   * One ACL rule is represented by the following 6-tuple.
+   */
   message Rule {
-    string src_ip = 1;    // Source IP block in CIDR. Wildcard if "".
-    string dst_ip = 2;    // Destination IP block in CIDR. Wildcard if "".
-    uint32 src_port = 3;  // Source port: 80. Wildcard if 0.
-    uint32 dst_port = 4;  // Destination port. Wildcard if 0.
-    bool established = 5; // Not implemented
-    bool drop = 6;        // Drop matched packets if true.
+    string src_ip = 1;    /// Source IP block in CIDR. Wildcard if "".
+    string dst_ip = 2;    /// Destination IP block in CIDR. Wildcard if "".
+    uint32 src_port = 3;  /// TCP/UDP source port. Wildcard if 0.
+    uint32 dst_port = 4;  /// TCP/UDP Destination port. Wildcard if 0.
+    bool established = 5; /// Not implemented
+    bool drop = 6;        /// Drop matched packets if true.
```

Comment:
Just to clarify -- if the default behavior of the ACL module is to drop, and the drop field is not set for a rule, does that mean that matching traffic is forwarded?